src/mapeo-project.js

```diff
@@ -733,24 +735,55 @@ export class MapeoProject extends TypedEmitter {
         return id
       })
       presetsWithRefs.push({
-        ...value,
-        iconId: iconName && iconNameToId.get(iconName),
-        fieldIds,
+        preset: {
+          ...value,
+          iconId: iconName && iconNameToId.get(iconName),
+          fieldIds,
+        },
+        name,
       })
     }
 
+    let presetPromises = []
+    for (let { preset, name } of presetsWithRefs) {
+      presetPromises.push(
+        this.preset.create(preset).then(({ docId }) => {
+          presetNameToId.set(name, docId)
+        })
+      )
+    }
+
+    await Promise.all(presetPromises)
+
+    const translationPromises = []
+    for (let { name, value } of config.translations()) {
+      let docIdRef
+      if (value.schemaNameRef === 'fields') {
+        docIdRef = fieldNameToId.get(name)
+      } else if (value.schemaNameRef === 'presets') {
+        docIdRef = presetNameToId.get(name)
+      }
```

Comment:
Maybe we could error if the `schemaNameRef` is something unexpected?

```suggestion
      } else {
        throw new Error(
          `${JSON.stringify(value.schemaNameRef)} must be "fields" or "presets"`
        )
```